trio/_sync.py

```diff
@@ -59,7 +65,15 @@ async def wait(self):
         if self._flag:
             await trio.lowlevel.checkpoint()
         else:
-            await self._lot.park()
+            task = _core.current_task()
+            self._tasks.add(task)
+            task.custom_sleep_data = self
```

Comment:
Why use custom sleep data here? Can't you just access `self._tasks` directly in the closure?

Reply:
good point

I've copied this code from ParkingLot, so I guess I'll apply the same simplification there

Reply:
`ParkingLot` needs `custom_sleep_data` b/c of the "requeueing" functionality that condition variables need -- see `ParkingLot.repark`

Reply:
I see, ParkingLot uses `custom_sleep_data` to support `repark()`